### PR TITLE
Automate swap partition setup for centos / ubuntu / debian servers

### PR DESCRIPTION
**Pull request by yashodhank**
I observed on many low memory VMs setup_frappe.sh fails because of no swap partition, on the same system if swap is available MariaDB and rest of setup insturctions executes without fail.

### CODE REVIEW

**Comment by anandpdoshi** [comment]
@yashodhank woah! 

Kudos to you. I find it very difficult to edit that shell script.

@pdvyas can you review the pull request?

---
PS: I guess it is time to modularise the setup script, maybe in python :v:

**Comment by yashodhank** [comment]
@anandpdoshi Sorry for the trouble it may have caused to you, I am improving it and trying to make it more modular, but at present tested on Ubuntu 14.04, Debian 7 and going to testing it on centos 6 and 7 shortly.

**Comment by yashodhank** [comment]
I observed on many low memory VMs setup_frappe fails because of no swap, on same system swap is available MariaDB does installs without any issues.

**Comment by yashodhank** [comment]
#### Ubuntu 14.04.2 LTS 
```
Welcome to Ubuntu 14.04.2 LTS (GNU/Linux 3.16.0-34-generic i686)

 * Documentation:  https://help.ubuntu.com/

  System information as of Tue Aug 25 23:00:59 UTC 2015

  System load:  0.12             Processes:           75
  Usage of /:   8.6% of 9.72GB   Users logged in:     0
  Memory usage: 12%              IP address for eth0: 104.245.37.21
  Swap usage:   0%

  Graph this data and manage this system at:
    https://landscape.canonical.com/

Last login: Tue Aug 25 23:01:00 2015 from 223.196.20.22
root@erpnext:~# rm -fr setup_frappe.sh; wget https://raw.githubusercontent.com/yashodhank/bench/dev/install_scripts/setup_frappe.sh
--2015-08-25 23:01:36--  https://raw.githubusercontent.com/yashodhank/bench/dev/install_scripts/setup_frappe.sh
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 199.27.76.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|199.27.76.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 19214 (19K) [text/plain]
Saving to: ‘setup_frappe.sh’

100%[==========================================================================================================================================>] 19,214      --.-K/s   in 0s      

2015-08-25 23:01:36 (139 MB/s) - ‘setup_frappe.sh’ saved [19214/19214]

root@erpnext:~# sudo bash setup_frappe.sh --setup-swap --setup-production
Installing for Ubuntu trusty x86
In case you encounter an error, you can post on https://discuss.frappe.io

Adding ubuntu mariadb repo
Step 1. No lock file, begin to create lock file and continue.
Step 2. Check this OS type.
This OS is ubuntu.
Step 3. Check the memory and swap.
Your swap is not enough, need to add swap.
Step 3. The swap partition setup successful.
Step 4. Begin to modify /etc/fstab.
Begin to modify /etc/fstab.
Step 5. Begin to adjust swappiness & vfs_cache_pressure
             total       used       free     shared    buffers     cached
Mem:           240        215         24          0          8        179
-/+ buffers/cache:         27        212
Swap:         2047          0       2047
All the swap setup related operations were completed. Returing to frappe installation.
Installing packages for Ubuntu. This might take time...
Installing wkhtmltopdf
 * Stopping MariaDB database server mysqld                                                                                                                                   [ OK ] 
 * Starting MariaDB database server mysqld                                                                                                                                   [ OK ] 
 * Checking for corrupt, not cleanly closed and upgrade needing tables.
Adding frappe user
Installing frappe-bench
Setting up first site

Frappe/ERPNext is installed successfully and is running on port 80.
Frappe password: hFtZgqkNFOizWjLR
MariaDB root password: iAU5SyQqnb37SDcW
Administrator password: w8CJLc74JG4JvZRp

The passwords are also stored at ~/frappe_passwords.txt
You can remove this file after making a note of the passwords.
```

**Comment by pdvyas** [comment]
Good effort!
Instead of lock file, let's just quit if swap exists already.. Also, there's distro detection already, I'd reuse that.

 I normally just do,
```
sudo fallocate -l 4G /swapfile
sudo chmod 600 /swapfile
sudo mkswap /swapfile
sudo swapon /swapfile
```

**Comment by pdvyas** [comment]
> PS: I guess it is time to modularise the setup script, maybe in python 

Not yet, not yet

**Comment by yashodhank** [comment]
@pdvyas please verify now.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,7 @@ Supported for CentOS 6, CentOS 7, Debian 7 and Ubuntu 12.04+
 Open your Terminal and enter:
 
 ```
-wget https://raw.githubusercontent.com/frappe/bench/master/install_scripts/setup_frappe.sh
+rm -fr setup_frappe.sh; wget https://raw.githubusercontent.com/frappe/bench/master/install_scripts/setup_frappe.sh
```
Comment (anandpdoshi) [inline README.md]: @yashodhank you need to remove this `rm -fr` command.

**Comment by anandpdoshi** [comment]
@yashodhank can you squash this pull request into a single commit?

**Comment by yashodhank** [comment]
I messed up with git merge with all experimental commands.. closing and creating new pull request.

**Pull request closed without merging**